### PR TITLE
Add Bgp config for neighbors in VRF

### PR DESCRIPTION
**Pull request by sojindal**
This config includes:
- Setup asn and router id for bgp process
- Add/Delete neighbors inside VRF
- Get all neighbors in a VRF
- Config to shutdown a neighbor

### CODE REVIEW

File: sonic_api.yaml [hunk 1]
```diff
@@ -1400,3 +1676,19 @@ definitions:
       ip_addr:
         type: string
         description: IP address of directly attached device to the switch
+  PeerPolicyTemplateEntry:
```
Comment (prsunny) [inline sonic_api.yaml:1679]: Where is this used? I don't see a usage

**Review by prsunny — APPROVED** [review]
lgtm, @lguohan , @sumukhatv to signoff before merge

**Review by lguohan — APPROVED** [review]

File: sonic_api.yaml [hunk 2]
```diff
@@ -1400,3 +1676,10 @@ definitions:
       ip_addr:
         type: string
         description: IP address of directly attached device to the switch
+  NeighborEntry:
+    type: object
+    properties:
+      neighbor_ip:
+        type: string
+      neighbor_as:
+        type: integer
```
Comment (gbmsft) [inline sonic_api.yaml:1685]: This needs to be unit.

File: sonic_api.yaml [hunk 3]
```diff
@@ -1255,6 +1255,282 @@ paths:
...
 #----------------------------------------------
+# Setup Global BGP process
+#----------------------------------------------
+  '/config/bgp/{asn}/{router_id}':
+    put:
+      summary: Setup Global BGP configurations on device
+      parameters:
+        - name: asn
+          in: path
+          type: integer
```
Comment (gbmsft) [inline sonic_api.yaml:1266]: uint

Comment (sojindal) [reply sonic_api.yaml:1266]: There doesn't seem to be a way to specify uint. Yaml has integer data type. As per my understanding, applications will need to implement the extra logic.

File: sonic_api.yaml (hunk 2 above)
Comment (sojindal) [reply sonic_api.yaml:1685]: Couldn't find any uint data type for yaml. It only has integers.
https://swagger.io/docs/specification/data-models/data-types/

**Pull request merged**